shacl/DESIGN.md

```diff
@@ -0,0 +1,111 @@
+# SHACL for Brick
+
+The SHACL Shapes Constraint Language validates RDF graphs against conditions (called "shapes"), which are expressed themselves as an RDF graph. This document discusses the application of SHACL to Brick. In this document we are specifically looking at writing SHACL shapes for validating Brick *models* (instances of buildings). Using SHACL to validate the Brick ontology deserves its own investigation.
+
+SHACL shapes for Brick should validate the following:
+
+- validating endpoints of Brick relationships (properties):
+  - check type of subject/object: are the `rdfs:range`, `rdfs:domain` properties respected?
+  - cardinality checks: with varying severity
+    - floors without rooms
+    - equipment missing certain points (e.g. thermostat should have a temperature sensor, at least one setpoint)
+    - all named individuals should have `rdf:type`
+- compatibility checking:
+  - all objects of `rdf:type` should be Brick classes (subclassof, etc)
+  - check for deprecated classes, relationships
+- idioms, semantic checks:
+  - these may be optional; users can "opt-in" to check for certain idioms
+  - equipment along a chain of `brick:feeds` should all operate on the same substance
+  - proper relationships exist between equipment: e.g. thermostat should have `controls` relationship to RTU
+
+## Getting Started
+
+Install `pyshacl` from pip.
+
+`generate_shacl.py` uses `bricksrc/` to generate a Shapes graph that is serialized to `shacl_test.ttl`.
+
+To validate a graph, use [pySHACL](https://github.com/RDFLib/pySHACL):
+
+```bash
+pyshacl -s shacl_test.ttl -m -e Brick.ttl -f human sample_graph.ttl
+```
+
+You may want to expand a graph before validating it. In that case, run
+
+```bash
+python expand_graph.py sample_graph.ttl
```

Comment:
We should probably remove this because the `expand_graph.py` script is gone; the inference should be handled by `pyshacl` directly if it is needed